### DISCUSSION TITLE
How the pkcs12_parse function supports the RC2-40-CBC algorithm in the legacy provider？

### DISCUSSION THREAD

**Discussion by Wormhole-design** [openssl/openssl#22654] [Q&A]
When we upgraded openssl from 1.1.1 to 3.0.9, a problem occurred in our business code. It is currently located that an error occurred in the PKCS12_parse function parsing the syslog certificate. The reason is that the RC2-40-CBC algorithm has been abandoned and placed in the legacy provider. In order to make the upgrade seamless for users, we cannot allow users to upgrade the syslog certificate. We can only make the PKCS12_parse function compatible with the RC2-40-CBC algorithm. How should I do this?

**Comment by paulidale** [comment]
You need to load the legacy provider to access such a pathetically weak and broken algorithm.
Read the `INSTALL.md` for instructions about this.
If that's not enough look at the `README-PROVIDERS.md` file.
The [migration guide](https://github.com/openssl/openssl/blob/master/doc/man7/ossl-guide-migration.pod) is also beneficial.

**Comment by levitte** [comment]
I'd say this is the quickest answer:

https://github.com/openssl/openssl/blob/22fa1602da91af2194997e0576582bb4f0cdd7e0/README-PROVIDERS.md#L84-L145
